solutions/cf-616D.mdx

```diff
@@ -0,0 +1,72 @@
+---
+id: cf-616D
+source: CF
+title: Longest k-Good Segment
+author: Sofia Yang
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/22712)
+
+<LanguageSection>
+
+<JavaSection>
+
+## Solution
+
+**Time Complexity:** $\mathcal{O}(N)$
+
+## Java Implementation
+
+```java
+public class KGood {
+	CodeSnip{Kattio}
+
+	public static int MAX = (int) 1e6;
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio();
+		int N = io.nextInt();
+		int K = io.nextInt();
+		int[] arr = new int[N];
+		int[] freq = new int[MAX];  // Frequency of each number in the window.
```

Comment:
uh why not make `freq` also just size `N` as well?

Reply:
In the [problem,](https://codeforces.com/problemset/problem/616/D) the values of the array `arr` aren't bounded between 1 and N, they're bounded between (0 ≤ arr[i] ≤ 10^6)

The frequency array `freq` contains the frequency of each number `arr[i]` , therefore, I think that the length of `freq[]` should be the maximum value in `arr[i]`.
And because I was lazy and new that the max value of `arr[i]` is 10^6, I set the length of `freq` to 10^6.

If you want, I could add a for loop to find the maximum value of arr[i] and then set that as the length of freq, I guess?

(Sorry if I'm being confusing, and feel free to correct me if I'm wrong )

Reply:
oof sorry i didn't read it clear enough
> If you want, I could add a for loop to find the maximum value of arr[i] and then set that as the length of freq, I guess?

this is what i'd do tbh (a oneliner would be `max = Arrays.stream(arr).max().getAsInt()`) but i'm not sure what the other people want

Reply:
Ok, I fixed it :)